packages/next/next-server/server/image-optimizer.ts

```diff
@@ -234,8 +234,15 @@ export async function imageOptimizer(
     const animate =
       ANIMATABLE_TYPES.includes(upstreamType) && isAnimated(upstreamBuffer)
     if (vector || animate) {
-      await writeToCacheDir(hashDir, upstreamType, expireAt, upstreamBuffer)
-      sendResponse(req, res, upstreamType, upstreamBuffer)
+      const etag = getHash([upstreamBuffer])
+      sendResponse(req, res, upstreamType, etag, upstreamBuffer)
+      await writeToCacheDir(
+        hashDir,
+        upstreamType,
+        expireAt,
+        etag,
+        upstreamBuffer
+      )
```

Comment:
Moved `etag` outside to avoid duplicate calculations, since it can increase CPU usage when the image is large. Also moved `sendResponse` earlier for a better TTFB.